h2o-genmodel-extensions/jgrapht/src/main/java/hex/genmodel/tools/JgraphtPrintMojo.java

```diff
@@ -0,0 +1,224 @@
+package hex.genmodel.tools;
+
+import com.mxgraph.layout.mxCompactTreeLayout;
+import com.mxgraph.layout.mxEdgeLabelLayout;
+import com.mxgraph.layout.mxIGraphLayout;
+import com.mxgraph.util.mxCellRenderer;
+import hex.genmodel.algos.tree.ConvertTreeOptions;
+import hex.genmodel.algos.tree.SharedTreeGraph;
+import hex.genmodel.algos.tree.SharedTreeSubgraph;
+import hex.genmodel.algos.tree.TreeBackedMojoModel;
+import org.jgrapht.ext.JGraphXAdapter;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.DirectedMultigraph;
+import org.jgrapht.io.*;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.Map;
+
+public class JgraphtPrintMojo extends PrintMojo implements MojoPrinter {
+
+    public static void main(String[] args) {
+        MojoPrinter mojoPrinter;
+        if ("png".equals(getFormat(args))){
+            mojoPrinter = new JgraphtPrintMojo();
+        } else {
+            mojoPrinter = new PrintMojo();
+        }
+        mojoPrinter.parseArgs(args);
+
+        // Run the main program
+        try {
+            mojoPrinter.run();
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.exit(2);
+        }
+        
+        // Success
+        System.exit(0);
+    }
+    
+    @Override
+    protected void validateArgs() {
+        if (genModel == null) {
+            System.out.println("ERROR: Must specify -i");
+            usage();
+        }
+    }
+
+    @Override
+    public void run() throws Exception {
+        validateArgs();
+     
+        if (genModel instanceof TreeBackedMojoModel){
+            TreeBackedMojoModel treeBackedModel = (TreeBackedMojoModel) genModel;
+            ConvertTreeOptions options = new ConvertTreeOptions().withTreeConsistencyCheckEnabled();
+            final SharedTreeGraph g = treeBackedModel.convert(treeToPrint, null, options);
+            printPng(g);
+        }
+        else {
+            System.out.println("ERROR: Unknown MOJO type");
+            System.exit(1);
+        }
+    }
+
+    private void printPng(SharedTreeGraph trees) throws IOException, ImportException {
+        Path outputDirectoryPath = Paths.get(outputFileName);
+        int numberOfTrees = trees.subgraphArray.size();
+        if (Files.exists(outputDirectoryPath)) {
+            Files.walk(outputDirectoryPath)
+                    .sorted(Comparator.reverseOrder())
+                    .map(Path::toFile)
+                    .forEach(File::delete);
```

Comment:
why do we need to delete everything in the output directory?

Reply:
In general, when we create output by PrintMojo, previous output is rewritten (existing behaviour). When we consider output directory as an output and it is left there from previous run, filled with .pngs and we want to generate new output with new run, on the same path, this was intended to preserve existing behaviour: rewrite existing output (=old directory with its contents) by new output (= new output directory with .pngs from actual run). if we want to preserve old content, I think this could be done in a way that we will move old content (if there is any) to archive directory, as discussed. What do you think?

Reply:
the output directory can be any directory, we do not care about there being other files and we should not presume anything about the contents of the directory, we only create it if does not exsits, we should definity remove everything, just overwrite the files we need to

Reply:
ok, so I understood it wrongly, will rework

Reply:
fixed